File: src/main/java/painpoints/Complex.java
package painpoints;

public class Complex {
    private double real;
    private double imaginary;

    public Complex(double real, double imaginary) {
        this.real = real;
        this.imaginary = imaginary;
    }

    public Complex add(Complex complex) {
        return new Complex(
                real + complex.real,
                imaginary + complex.imaginary
        );
    }

    public static void main(String[] args) {
        Complex a = new Complex(1, 2),
                b = new Complex(3,4);
        System.out.println(a.add(b));
    }
}

File: src/main/kotlin/solve/Complex.kt
package solve

class Complex(val real: Double, val imaginary: Double) {
    operator fun plus(c: Complex) =
            Complex(real + c.real, imaginary + c.imaginary)
}

fun main(args: Array<String>) {
    val a = Complex(1.0, 2.0)
    val b = Complex(3.0, 4.0)
    println(a + b)
}
